### DISCUSSION TITLE
How to use $env variables as entities?

### DISCUSSION THREAD

**Discussion by tamueller** [zachowj/node-red-contrib-home-assistant-websocket#844] [General]
Hi,
I want to create a subflow that does complex logic that should be reused for different entities. I want to configure the entity in the subflow properties. I can access the properties of the subflow in other places, but I'm not sure how to choose the entity of e.g. a "call-service-node" with env-variables.
The subflow property is called "id" and I tried `$env('id')` `$id`, and` {{env.id}} `but I always get: `"Call-service error. not a valid value for dictionary value @ data['entity_id']"`

Is it possible to choose the entity during runtime somehow, if so, how?

**Comment by zachowj** [comment]
You could add it to the data field.

```
{
  "entity_id": $env('id')
}
```

or build an input using a change/function node and pass it to the node

<https://zachowj.github.io/node-red-contrib-home-assistant-websocket/node/call-service.html#input>

**Comment by tamueller** [comment]
Thanks, I didn't know that you could override things with data or message - that's pretty helpful.
